python/hsfs/feature_view.py

```diff
@@ -884,6 +1473,84 @@ def train_validation_test_split(
         """
         Get training data from feature groups.
 
+        !!! example 
+            ```python
+            # get feature store instance 
+            fs = ...
+
+            # get feature view instance 
+            feature_view = fs.get_feature_view(...)
+
+            # get training data
+            X_train, X_val, X_test, y_train, y_val, y_test = feature_view.train_validation_test_split(
+                validation_size=0.3,
+                test_size=0.2
+            )
+            ```
+
+        !!! example "Time Series split"
+            ```python
+            # get feature store instance 
+            fs = ...
+
+            # get feature view instance 
+            feature_view = fs.get_feature_view(...)
+
+            # set up dates
+            start_time_train = '2017-01-01 00:00:01'
+            end_time_train = '2018-02-01 23:59:59'
+
+            start_time_val = '2018-02-02 23:59:59'
+            end_time_val = '2019-02-01 23:59:59'
+            
+            start_time_test = '2019-02-02 23:59:59'
+            end_time_test = '2020-02-01 23:59:59'
+            # you can also pass dates as datetime objects
+
+            # get training data
+            X_train, X_val, X_test, y_train, y_val, y_test = feature_view.train_validation_test_split(
+                validation_size=0.3,
+                test_size=0.2,
+                train_start=start_time_train,
+                train_end=end_time_train,
+                validation_start=start_time_val,
+                validation_end=end_time_val,
+                test_start=start_time_test,
+                test_end=end_time_test
+            )
+            ```
+        !!! warning "Label Encoder can fail while creating training splits. Feature view is just a logical union of feature groups which stores a metadata. While creating dataset splits - label encoder fits on train split. There may be unique values in the test or validation data. As a result Label Encoder fails in handling unique unfitted values."            
+            ```python
+            import pandas as pd
+
+            df = pd.DataFrame({
+                'category_col':['category_a','category_b','category_c','category_d'],
+                'numeric_col': [40,10,60,40]
+            })
+
+            feature_group = fs.get_or_create_feature_group(
+                name='feature_group_name',
+                version=1,
+                primary_key=['category_col']
+            )
+
+            feature_group.insert(df)
+
```

Comment:
can you please also add example and explanation of `feature_group.insert(df, write_options={"wait_for_job" : False})`